components/tikv_util/src/metrics/threads_linux.rs

```diff
@@ -466,11 +474,123 @@ impl TidRetriever {
     }
 }
 
+lazy_static::lazy_static! {
+    pub static ref THREAD_NAME_HASHMAP: Mutex<HashMap<Pid, String>> = Mutex::new(HashMap::default());
+}
+
+fn add_thread_name_to_map() {
+    if let Some(name) = std::thread::current().name() {
+        let tid = thread::thread_id();
+        THREAD_NAME_HASHMAP
+            .lock()
+            .unwrap()
+            .insert(tid, name.to_string());
+        debug!("tid {} thread name is {}", tid, name);
+    }
+}
+
+fn remove_thread_name_from_map() {
+    let tid = thread::thread_id();
+    THREAD_NAME_HASHMAP.lock().unwrap().remove(&tid);
+}
+
+impl StdThreadBuildWrapper for std::thread::Builder {
+    fn spawn_wrapper<F, T>(self, f: F) -> Result<std::thread::JoinHandle<T>>
+    where
+        F: FnOnce() -> T,
+        F: Send + 'static,
+        T: Send + 'static,
+    {
+        #[allow(clippy::disallowed_methods)]
+        self.spawn(|| {
+            add_thread_name_to_map();
+            let res = f();
+            remove_thread_name_from_map();
+            res
+        })
+    }
+}
+
+impl<T: PoolTicker> ThreadBuildWrapper for YatpPoolBuilder<T> {
```

Comment:
Even the `Pool` is defined by the `tikv_util`, why bother to add an extra layer? For example, why not just put the hooks to runner directly?

Reply:
I just modify the `after_start` and `before_stop` of `YatpPoolBuilder`, instead of implementing trait for it.